msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -66,66 +107,270 @@ private void initializeSharedPreferenceFileManager(@NonNull final Context contex
+            if (newAccount != null) {
+                callback.onAccountChanged(null, newAccount);
+            }
+        } else if (localAccount.getId() != newAccount.getId()) {
+            callback.onAccountChanged(localAccount, newAccount);
+        }
+
+        persistCurrentAccount(newAccountRecords);
+        callback.onAccountLoaded(newAccount);
+    }
+
+
+    @Override
+    public void signIn(@NonNull Activity activity,
+
+                @NonNull String[] scopes,
+                @NonNull AuthenticationCallback callback) {
+        acquireToken(
+                activity,
+                new String[]{"user.read"},
+                null, // account
+                null, // uiBehavior
+                null, // extraQueryParams
+                scopes, // extraScopes
+                null, // authority
+                callback,
+                null, // loginHint
+                null // claimsRequest
+        );
+    }
+
     @Override
-    public void removeCurrentAccount(@NonNull final TaskCompletedCallbackWithError<Boolean, Exception> callback) {
+    protected ILocalAuthenticationCallback getLocalAuthenticationCallback(final AuthenticationCallback authenticationCallback) {
+
+        return new ILocalAuthenticationCallback() {
+
+            @Override
+            public void onSuccess(ILocalAuthenticationResult localAuthenticationResult) {
+
+                //Get Local Authentication Result then check if the current account is set or not
+                MultiTenantAccount newAccount =  getAccountFromICacheRecordList(localAuthenticationResult.getCacheRecordWithTenantProfileData());
+
+                if(didExistingCurrentAccountChange(newAccount)){
+                    //Throw on Error with UserMismatchException
+                    authenticationCallback.onError(new MsalClientException(MsalClientException.CURRENT_ACCOUNT_MISMATCH));
+                    return;
+                }else{
+                    persistCurrentAccount(localAuthenticationResult.getCacheRecordWithTenantProfileData());
+                }
+
+                IAuthenticationResult authenticationResult = AuthenticationResultAdapter.adapt(localAuthenticationResult);
+                authenticationCallback.onSuccess(authenticationResult);
+            }
+
+            @Override
+            public void onError(BaseException exception) {
+                MsalException msalException = MsalExceptionAdapter.msalExceptionFromBaseException(exception);
+                authenticationCallback.onError(msalException);
+            }
+
+            @Override
+            public void onCancel() {
+                authenticationCallback.onCancel();
+            }
+        };
+    }
+
+    private boolean didExistingCurrentAccountChange(MultiTenantAccount newAccount){
+        MultiTenantAccount persistedAccount = getPersistedCurrentAccount();
+        return (persistedAccount != null || newAccount.getId() != persistedAccount.getId()) ? true : false;
```

Comment:
🐞 String comparison using `!=` -- should be `... || !newAccount.getId().equals(persistedAccount.getId())`

But I think there's another issue here - if `persistedAccount != null` the right-side of this expression will never be evaluated. Should this be `if (null == persistedAccount)` ?

Reply:
Fixed.

Reply:
Was actually missing the null check for newAccount as well.  Updated signature to mark as nullable since i'm now reusing this logic from somewhere else.